src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -60,7 +61,10 @@ public override void OnEventWritten(string name, object? payload)
                     if (activity == null)
                     {
                         // There is no listener or it decided not to sample the current request.
-                        return;
+                        activity = new Activity(SqlActivitySourceHelper.MeterName);
```

Comment:
unsure why is `Activity` created this way here?

Reply:
Instead of creating an activity, could you incorporate the approach you used in your [previous PR](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1989/files#diff-339e87239afadd02c6b5e405f59714f3f538a1086db13d4ed983c4850baab098R58-R66)?

The idea being:
* When tracing is disabled use the stopwatch when recording the duration.
* When tracing is enabled use the activity when recording the duration so that the two are in sync.

Reply:
+1. "Activity" is a bit heavy heap object, AsyncLocal-Stopwatch is much lighter.